tests/cmake/tpls.cmake

```diff
@@ -0,0 +1,170 @@
+    find_package(MPI REQUIRED)
+    include_directories(${MPI_CXX_INCLUDE_PATH})
+  endif()
+endif()
+
+
+if(PRESSIO_ENABLE_TPL_BLAS OR PRESSIO_ENABLE_TPL_TRILINOS)
+    # check if BLAS_ROOT is specified
+    if (NOT ${BLAS_ROOT})
+      message("")
+      message(FATAL_ERROR "BLAS_ROOT not speificed, terminating")
+      message("Make sure you set the BLAS_ROOT env var")
+    endif()
+
+    cmake_policy(SET CMP0074 NEW)
+    find_package(BLAS REQUIRED)
+    link_libraries(${BLAS_LIBRARIES})
+    message("")
+endif()
+
+if(PRESSIO_ENABLE_TPL_LAPACK)
+  if(PRESSIO_ENABLE_UNIT_TESTS OR PRESSIO_ENABLE_TESTS)
+    # check if LAPACK_ROOT is specified
+    if (NOT ${LAPACK_ROOT})
+      message("")
+      message(FATAL_ERROR "LAPACK_ROOT not speificed, terminating")
+      message("Make sure you set the LAPACK_ROOT env var")
+    endif()
+
+    cmake_policy(SET CMP0074 NEW)
+    find_package( LAPACK REQUIRED )
+    link_libraries(${LAPACK_LIBRARIES})
+    message("LAPLIBS=${LAPACK_LIBRARIES}")
+  endif()
+endif()
+
+
+if(PRESSIO_ENABLE_TPL_EIGEN)
+  if(PRESSIO_ENABLE_UNIT_TESTS OR PRESSIO_ENABLE_TESTS)
+
+    if(NOT EIGEN_INC_DIR AND NOT EIGEN_INCLUDE_DIR)
+      message(FATAL_ERROR
+	"I cannot find the Eigen headers. Please reconfigure with:
+      	-DEIGEN_INC_DIR=<full-path-to-headers>
+      	or
+      	-DEIGEN_INCLUDE_DIR=<full-path-to-headers>
+      ")
+    endif()
+
+    if(NOT EIGEN_INC_DIR AND EIGEN_INCLUDE_DIR)
+      set(EIGEN_INC_DIR ${EIGEN_INCLUDE_DIR})
+    endif()
+
+    include_directories(${EIGEN_INC_DIR})
+  endif()
+endif()
+
+
+if(PRESSIO_ENABLE_TPL_KOKKOS)
+  if(PRESSIO_ENABLE_UNIT_TESTS OR PRESSIO_ENABLE_TESTS)
+
+    # # when trilinos is also enabled it links kokkos too, see tplTrilinos.cmake
+    if(NOT PRESSIO_ENABLE_TPL_TRILINOS)
+      # if kokkos is used as standalone lib, then we are more specific
+      # user needs to defined: KOKKOS_ROOT_DIR and KOKKOS_KERNELS_ROOT_DIR
+      if (NOT KOKKOS_ROOT OR NOT KOKKOS_KERNELS_ROOT)
+	message(FATAL_ERROR "Missing KOKKOS_ROOT. KOKKOS needs:
+          -D KOKKOS_ROOT=<full-path-to-kokkos-installation>
+          -D KOKKOS_KERNELS_ROOT=<full-path-to-kokkos-kernels-installation>
+          ")
+      endif()
+
+      set(KOKKOS_LIB_NAMES kokkoscontainers kokkoscore kokkoskernels)
+
+      include_directories(${KOKKOS_ROOT}/include ${KOKKOS_KERNELS_ROOT}/include)
+
+      link_directories(${KOKKOS_ROOT}/lib ${KOKKOS_ROOT}/lib64
+	${KOKKOS_KERNELS_ROOT}/lib ${KOKKOS_KERNELS_ROOT}/lib64)
+
+      link_libraries(${KOKKOS_LIB_NAMES})
+    endif()
+
+  endif()
+endif()
+
+
+if(PRESSIO_ENABLE_TPL_TRILINOS)
+  if(PRESSIO_ENABLE_UNIT_TESTS OR PRESSIO_ENABLE_TESTS)
+    if (NOT TRILINOS_ROOT)
+      message(FATAL_ERROR
+	"You enabled PRESSIO_ENABLE_TPL_TRILINOS but did not set TRILINOS_ROOT.
+        Please reconfigure with:
+          -D TRILINOS_ROOT=<full-path-to-trilinos-install>
+          ")
+    endif()
+
+    set(TRILINOS_LIB_NAMES kokkosalgorithms
```

Comment:
basically here we just need to use cmake find package

Reply:
makes sense?

Reply:
here is what is enabled when trilinos is on: https://github.com/Pressio/pressio/blob/develop/cmake/tpls/tplTrilinos.cmake

currently in develop

Reply:
Switched to `find_package`:
https://github.com/Pressio/pressio/blob/0dc3d12e2eab8fd31dd3fc07b0113f35c166966a/tests/cmake/options.cmake#L53-L59

Reply:
Up for a debate whether we want to search for subpackages.